src/renderer/src/providers/ChatProvider.jsx

```
@@ -1893,19 +1910,58 @@ const useChatStore = create((set, get) => ({
         break;
       case "cosmetic.create":
-        useCosmeticsStore?.getState()?.addCosmetics(body);
+        console.log(
+          `[ChatProvider] Applying cosmetic catalog update for ${chatroomId ?? 'all chatrooms'}`,
+          {
+            badges: body?.badges?.length,
+            paints: body?.paints?.length,
+          },
+        );
+        const cosmetics = useCosmeticsStore?.getState()?.addCosmetics;
+        if (cosmetics) {
```

Biome (2.1.2)

[error] Other switch clauses can erroneously access this declaration.
Wrap the declaration in a block to restrict its access to the switch clause.

The declaration is defined in this switch clause:

Safe fix: Wrap the declaration in a block.


(lint/correctness/noSwitchDeclarations)